pkg/overlay/server_test.go

```diff
@@ -65,3 +67,142 @@ func TestServer(t *testing.T) {
 		}
 	}
 }
+
+func TestNewNodeFiltering(t *testing.T) {
+	ctx := testcontext.New(t)
+	defer ctx.Cleanup()
+
+	planet, err := testplanet.New(t, 1, 4, 1)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer ctx.Check(planet.Shutdown)
+
+	planet.Start(ctx)
+	// we wait a second for all the nodes to complete bootstrapping off the satellite
+	time.Sleep(2 * time.Second)
+
+	satellite := planet.Satellites[0]
+
+	for i, tt := range []struct {
+		name                  string
+		newNodeAuditThreshold int64
+		newNodePercentage     float64
+		requestedNodeAmt      int64
+		expectedResultLength  int
+		reputableNodes        int
+	}{
+		{
+			name:                  "case: fewer than required reputable nodes",
+			requestedNodeAmt:      4,
+			reputableNodes:        3,
+			expectedResultLength:  3,
+			newNodeAuditThreshold: 1,
+		},
+		{
+			name:                  "case: more than required reputable nodes",
+			requestedNodeAmt:      2,
+			reputableNodes:        3,
+			expectedResultLength:  2,
+			newNodeAuditThreshold: 1,
+		},
+		{
+			name:                  "case: zero reputable nodes found, only new nodes",
+			requestedNodeAmt:      2,
+			reputableNodes:        0,
+			expectedResultLength:  2,
+			newNodeAuditThreshold: 1,
+		},
+		{
+			name:              "case: fewer than required new nodes",
+			requestedNodeAmt:  2,
+			reputableNodes:    3,
+			newNodePercentage: 0.5,
+			// this gives extra reputable instead
+			expectedResultLength:  3,
+			newNodeAuditThreshold: 1,
+		},
+		{
+			name:                  "case: more than required new nodes",
+			requestedNodeAmt:      2,
+			reputableNodes:        2,
+			newNodePercentage:     0.5,
+			expectedResultLength:  3,
+			newNodeAuditThreshold: 1,
+		},
+		{
+			// todo(nat): fix nodes length issue
+			name:                  "case: zero new nodes found, only reputable nodes",
+			requestedNodeAmt:      3,
+			reputableNodes:        3,
+			newNodePercentage:     0.5,
+			expectedResultLength:  4,
+			newNodeAuditThreshold: 1,
+		},
+		{
+			name:                  "case: exactly the required amount of new and reputable nodes returned",
+			requestedNodeAmt:      1,
+			reputableNodes:        1,
+			newNodePercentage:     1,
+			expectedResultLength:  2,
+			newNodeAuditThreshold: 1,
+		},
+		{
+			name:              "case: low percentage of new nodes",
+			requestedNodeAmt:  3,
+			reputableNodes:    1,
+			newNodePercentage: 0.01,
+			// todo(nat): expect this result to be 1
+			expectedResultLength:  3,
+			newNodeAuditThreshold: 1,
+		},
+		{
+			name:                  "case: high percentage of new nodes",
+			requestedNodeAmt:      1,
+			reputableNodes:        1,
+			newNodePercentage:     3,
```

Comment:
Should we allow `newNodePercentage` to be greater than 1?

Reply:
that's a good point even with the new way of calculating the node results. because then it's like 3x the requested nodes in addition to the requested amount, which could get really big.